Shared/Samples/Create load report/CreateLoadReportView.Model.swift

```diff
@@ -0,0 +1,283 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension CreateLoadReportView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    @MainActor
+    class Model: ObservableObject {
+        // MARK: Properties
+        /// The utility network for this sample.
+        private let utilityNetwork = UtilityNetwork(url: .utilityNetwork)
+        
+        /// The initial conditional expression.
+        private var initialExpression: UtilityTraceConditionalExpression?
+        
+        /// The trace parameters for creating load reports.
+        private var traceParameters: UtilityTraceParameters?
+        
+        /// The network attributes for the comparison.
+        private var phasesNetworkAttribute: UtilityNetworkAttribute?
+        
+        /// A list of possible phases populated from the network's attributes.
+        /// By default, they are not included in the load report.
+        @Published private(set) var excludedPhases = [CodedValue]()
+        
+        /// A list of phases that are included in the load report.
+        @Published private(set) var includedPhases = [CodedValue]()
+        
+        /// A list of possible phases populated from the network's attributes.
+        private var allPhases = [CodedValue]()
+        
+        /// The phase summaries in the load report.
+        private var summaries = [CodedValue: PhaseSummary]()
+        
+        /// A Boolean value indicating if the run button is enabled.
+        @Published private(set) var runEnabled = false
+        
+        /// The status text to display to the user.
+        @Published private(set) var statusText: String?
+        
+        /// A Boolean value indicating if the sample is authenticated.
+        private var isAuthenticated: Bool {
+            !ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials.isEmpty
+        }
+        
+        // MARK: Methods
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async throws {
+            try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+            try await loadUtilityNetwork()
+        }
+        
+        /// Loads the utility network.
+        private func loadUtilityNetwork() async throws {
+            statusText = "Loading utility network…"
+            try await utilityNetwork.load()
+            statusText = nil
```

Comment:
Would the function observe load status changes to set the status text?

Reply:
No. It would be something that just took a closure and reset the status at the end of the closure.  However, it might not carry its own weight because it's easy to use a defer statement.